sdks/python/apache_beam/io/avroio_it_test.py

```diff
@@ -132,73 +128,54 @@ def batch_indices(start):
-        | 'create-batches' >> Create(batches) \
-        | 'expand-batches' >> FlatMap(batch_indices) \
-        | 'create-records' >> Map(record)
-
-    fastavro_output = '/'.join([self.output, 'fastavro'])
-    avro_output = '/'.join([self.output, 'avro'])
-
-    # pylint: disable=expression-not-assigned
-    records_pcoll \
-    | 'write_fastavro' >> WriteToAvro(
-        fastavro_output,
-        parse_schema(json.loads(self.SCHEMA_STRING)),
-        use_fastavro=True
-    )
-
-    # pylint: disable=expression-not-assigned
-    records_pcoll \
-    | 'write_avro' >> WriteToAvro(
-        avro_output,
-        Parse(self.SCHEMA_STRING),
-        use_fastavro=False
-    )
-
-    result = self.test_pipeline.run()
-    result.wait_until_finish()
-    assert result.state == PipelineState.DONE
-
-    fastavro_read_pipeline = TestPipeline(is_integration_test=True)
-
-    fastavro_records = \
-        fastavro_read_pipeline \
-        | 'create-fastavro' >> Create(['%s*' % fastavro_output]) \
-        | 'read-fastavro' >> ReadAllFromAvro(use_fastavro=True) \
-        | Map(lambda rec: (rec['number'], rec))
-
-    avro_records = \
-        fastavro_read_pipeline \
-        | 'create-avro' >> Create(['%s*' % avro_output]) \
-        | 'read-avro' >> ReadAllFromAvro(use_fastavro=False) \
-        | Map(lambda rec: (rec['number'], rec))
-
-    def check(elem):
-      v = elem[1]
-
-      def assertEqual(l, r):
-        if l != r:
-          raise BeamAssertException('Assertion failed: %s == %s' % (l, r))
-
-      assertEqual(v.keys(), ['avro', 'fastavro'])
-      avro_values = v['avro']
-      fastavro_values = v['fastavro']
-      assertEqual(avro_values, fastavro_values)
-      assertEqual(len(avro_values), 1)
-
-    # pylint: disable=expression-not-assigned
-    {
-        'avro': avro_records,
-        'fastavro': fastavro_records
-    } \
-    | CoGroupByKey() \
-    | Map(check)
-
-    self.addCleanup(delete_files, [self.output])
-    fastavro_read_pipeline.run().wait_until_finish()
-    assert result.state == PipelineState.DONE
+    _ = (write_pipeline
+         | 'CreateBatches' >> Create(batches)
+         | 'ExpandBatches' >> FlatMap(batch_indices)
+         | 'CreateRecords' >> Map(record)
+         | 'WriteAvro' >> WriteToAvro(files_output,
+                                      self.SCHEMA,
+                                      file_name_suffix=".avro",
+                                      use_fastavro=self.use_fastavro)
+        )
+
+    write_result = write_pipeline.run()
+    write_result.wait_until_finish()
+    assert write_result.state == PipelineState.DONE
+
+    def check(x, mean):
+      if not x == mean:
+        raise BeamAssertException('Assertion failed: %s == %s' % (x, mean))
+
+    _ = (read_pipeline
+         | "ReadAvro" >> ReadFromAvro(files_output + '*')
+         | 'ExtractNumber' >> Map(lambda x: x['number'])
+         | 'CalculateMean' >> Mean.Globally()
+         | 'MeanIsCorrect' >> Map(
```

Comment:
This test evaluates correctness only of a value derived from 'number' field. Can we instead compare that all attributes of the record are written and read correctly? Also, comparing float numbers with `==` is not reliable due to a lack of precision in the float numbers representation.

Reply:
resolved in new pr iteration.